### PR TITLE
fix a potential deadlock

### PR DESCRIPTION
**Pull request by charlesxsh**
#### What type of PR is this?
/kind bug

#### What this PR does / why we need it:
This PR tries to fix a potential deadlock issue. If line 247 happened first, line 229 or 231 will be blocked forever.

#### Which issue(s) this PR fixes:

Fixes #

#### Special notes for your reviewer:
Similar issue/fix: https://github.com/kubernetes/kubernetes/pull/83925

#### Does this PR introduce a user-facing change?

```release-note

```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: cmd/kube-apiserver/app/testing/testserver.go [hunk 1]
```diff
@@ -222,7 +222,7 @@ func StartTestServer(t Logger, instanceOptions *TestServerInstanceOptions, custo
 		return result, fmt.Errorf("failed to create server chain: %v", err)
 	}
 
-	errCh := make(chan error)
+	errCh := make(chan error, 1)
```
Comment (wojtek-t) [inline cmd/kube-apiserver/app/testing/testserver.go:225]: I don't think there is a risk of deadlock - putting into errCh is happening in a separate goroutine so everything seems fine to me.

**Comment by deads2k** [comment]
Any update here of whether this was actually observed or just theoretical?  I don't think the code deadlocks, but the gofunc starting the server could block on error if the intervening code somehow failed (low risk).

Similar to wojtek, this change doesn't appear to break anything, but I'd like to know how we got here.

File: cmd/kube-apiserver/app/testing/testserver.go (hunk 1 above)
Comment (charlesxsh) [reply cmd/kube-apiserver/app/testing/testserver.go:225]: The deadlock is only for child goroutine (line 226) and it only happened if line 247 got selected. This fix does not change any test behavior but simply allow proper exit of the child goroutine if line 247 got selected. Similar to #102635 and #83925.

**Comment by charlesxsh** [comment]
Hi there, kindly ask do we have any update on this PR? If nothing wrong, could this PR get merged?

**Comment by charlesxsh** [comment]
@wojtek-t Hi there, kindly ask do we have any update on this PR?

**Comment by songlh** [comment]
@deads2k If poll.wait() returns without receiving any messages from channel errCh, the child goroutine (created at line 226) is leaked at the sending operation on channel errCh. We observed the leaked goroutine in our testing.

**Comment by dims** [comment]
waiting on review from @deads2k

**Pull request closed without merging**